src/board.py

```diff
@@ -3,23 +3,45 @@
 
 class Board:
     [WEST, NORTH, EAST, SOUTH] = ["w", "n", "e", "s"]
+    # TODO check out of bounds obstacles
     DEFAULT_OBSTACLES = [(0, 3), (1, 1), (1, 5), (2, 2),
                          (2, 4), (3, 0), (3, 6), (4, 2), (4, 4), (5, 1), (5, 5), (6, 3)]
+    DEFAULT_EXITS = [(1, -1), (7, 1), (-1, 5), (5, 7), (7, 2)]
+    DEFAULT_PUCKS = [(3, 3)]
+    PUCK_EXITS = 'puck_exit'
 
-    def __init__(self, width=7, height=7, obstacles=DEFAULT_OBSTACLES):
+    def __init__(self, width=7, height=7, obstacles=DEFAULT_OBSTACLES, exits=DEFAULT_EXITS, pucks=DEFAULT_PUCKS):
         self.WIDTH = width
         self.HEIGHT = height
         self.graph = nx.DiGraph()
         self.__initialize_empty_board()
-        self.__initialize_pucks()
+        self.__initialize_pucks(pucks)
+        self.__initialize_exits(exits)
         self.__initialize_obstacles(obstacles)
 
-    def __initialize_pucks(self):
-        self.graph.nodes[(self.WIDTH//2, self.HEIGHT//2)]["puck"] = 'O'
+    def __initialize_pucks(self, pucks):
+        for puck in pucks:
+            self.__add_puck(puck)
 
     def __initialize_obstacles(self, obstacles):
+        # TODO mention this in the blog note?
         self.graph.remove_nodes_from(obstacles)
 
+    def __initialize_exits(self, exits):
+        for (i, j) in exits:
+            self.graph.add_node((i, j), exit=True)
```

Comment:
Maybe ensure that exits can't be in corners 
like (-1,-1) or (WIDTH, HEIGHT)
Because if you set some, it will create extra weird nodes to connect them.

Reply:
Yes, and beside that, it make me think that I have to add a condition to check that we are really creating an exit on the borders and not in the middle of the board...

Reply:
you're right